src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/ioc/CreateCitizenClaimCallbackHandlerTest.java

```diff
@@ -0,0 +1,216 @@
+        given(responseDeadlineCalculator.calculateResponseDeadline(any())).willReturn(RESPONSE_DEADLINE);
+        given(issueDateCalculator.calculateIssueDay(any())).willReturn(ISSUE_DATE);
+
+        UserDetails userDetails = SampleUserDetails.builder().withRoles("citizen").build();
+        given(userService.getUserDetails(AUTHORISATION_TOKEN)).willReturn(userDetails);
+    }
+
+    @Test
+    public void shouldAddFieldsOnCaseWhenCallbackIsSuccessful() throws Exception {
+        payment = paymentBuilder.status(SUCCESS).build();
+
+        given(paymentsService
+            .retrievePayment(
+                eq(AUTHORISATION_TOKEN),
+                any(Claim.class)))
+            .willReturn(payment);
+
+        MvcResult mvcResult = makeRequestAndRespondWithSuccess(CallbackType.ABOUT_TO_SUBMIT.getValue())
+            .andExpect(status().isOk())
+            .andReturn();
+        Map<String, Object> responseData = deserializeObjectFrom(
+            mvcResult,
+            AboutToStartOrSubmitCallbackResponse.class
+        ).getData();
+
+        List<Map<String, Object>> respondents = (List<Map<String, Object>>) responseData.get("respondents");
+        Map<String, Object> defendant = (Map<String, Object>) respondents.get(0).get("value");
+
+        assertThat(responseData).contains(
+            entry("paymentStatus", SUCCESS.toString()),
+            entry("issuedOn", ISSUE_DATE.toString()),
+            entry("previousServiceCaseReference", REFERENCE_NO)
+        );
+
+        assertThat(defendant).contains(entry("responseDeadline", RESPONSE_DEADLINE.toString()));
+    }
+
+    @Test
+    public void shouldAddFieldsOnCaseWhenCallbackIsSuccessfulButWithErrors() throws Exception {
+        payment = paymentBuilder.status(FAILED).build();
+
+        given(paymentsService
+            .retrievePayment(
+                eq(AUTHORISATION_TOKEN),
+                any(Claim.class)))
+            .willReturn(payment);
+
+        MvcResult mvcResult = makeRequestAndRespondWithError(CallbackType.ABOUT_TO_SUBMIT.getValue())
+            .andExpect(status().isOk())
+            .andReturn();
+
+        List<String> responseData = deserializeObjectFrom(
+            mvcResult,
+            AboutToStartOrSubmitCallbackResponse.class
+        ).getErrors();
+
+        assertThat(responseData).contains("Payment not successful");
+    }
+
+    @Test
+    public void shouldCallClaimIssuePostOperationsWhenSubmittedCallbackIsSuccessful() throws Exception {
+
+        given(userService.getUser(anyString())).willReturn(new User(BEARER_TOKEN,
+            SampleUserDetails.builder().build()));
+
+        MvcResult mvcResult = makeRequestAndRespondWithError(CallbackType.SUBMITTED.getValue())
+            .andExpect(status().isOk())
+            .andReturn();
+
+        SubmittedCallbackResponse response = deserializeObjectFrom(
+            mvcResult,
+            SubmittedCallbackResponse.class
+        );
+
+        verify(eventProducer, times(1))
```

Comment:
`times(1)` can be `once()` and is anyway redundant. Keep it only if you think it improves test clarity (which I personally don't).

Reply:
eventProducer should be called only once for post operations. I would like test to say that explicitly even if its just for info.